### DISCUSSION TITLE
500-as hibák (INTERNAL_SERVER_ERROR)

### DISCUSSION THREAD

**Discussion by sinick2** [nav-gov-hu/M2M#33] [Általános]
Tisztelt Fejlesztők!

Tesztrendszerben "createDocument" (REST API) végpont használata során előfordul, hogy 500-as hibát kapok a lenti válasszal. Hogyan készüljek? Az ilyen jellegű 500-as hibákra lesz javítás?

Köszönöm: sinick2

`{"type":"https://m2m.nav.gov.hu/technical-exception","title":"Technical exception occured in system: EBEV documentFileId: d4863763-a105-4595-8628-c5f9dba15b82","status":500,"detail":"feign.FeignException$InternalServerError: [500 Internal Server Error] during [POST] to [http://m2m-bizonylat-eloellenorzes/eloellenorzes] [BizonylatEloellenorzesServiceApi#bizonylatEloellenorzes(String,String,Mono)]: [{\"timestamp\":\"2024-10-22T12:16:22.872+00:00\",\"status\":500,\"error\":\"Internal Server Error\",\"trace\":\"jakarta.xml.ws.soap.SOAPFaultException: Connection prematurely closed BEFORE response\\n\\tat org.apach... (17375 bytes)]","instance":"/NavM2mDocument/documentService/Document","timestamp":"2024-10-22T12:16:22.878249243Z","application":"core-bizonylat","errorCategory":"Generic"}`

**Comment by sinick2** [comment]
Tisztelt Fejlesztők!

Most a "getFileStatus" (REST API) végpont adott 500-as hibát a tesztrendszerben. Folyamatot lentebb ismertetem. Ha a 2. pont "getFileStatus"-t többször ismétlem, akkor egyszer csak "PASSED" lesz. (Gondolom a folyamat valójában "WAITING" alatt van.)

Köszönöm: sinick2

1. Az "addFile" sikeres volt. Válasz: `{"resultMessage":null,"fileId":"42ba2372-e6d8-4083-999b-37634971b849","result_code":"UPLOAD_SUCCESS"}`
2. A "getFileStatus" státusza 500. Válasz (response) nincs.
3. A "createDocument" nem volt sikeres. Válasz: 
`{"resultMessage":"A következő azonosítóval rendelkező nyomtatvány vírusellenőrzése még nem fejeződött be: 42ba2372-e6d8-4083-999b-37634971b849","documentStatus":"PREVALIDATION_ERROR","resultCode":"UNSUCCESSFUL_VALIDATION"}`
4. A "getDocumentStatus" nem volt sikeres. Válasz: `{"resultMessage":null,"documentStatus":null,"documentFileId":"42ba2372-e6d8-4083-999b-37634971b849","errors":null,"arrivalNumber":null,"resultCode":"UNKNOWN_FILE_ID"}`

**Comment by Csonti-Piller** [comment]
A megjelölt esetre javítással készülünk, de ettől függetlenül javasoljuk az 500-as hibakódra való felkészülést, viszont az minden esetben olyan hibajelenséget takar, amely bejelentésre szorul - átlagos, normális üzemben nem elvárt és nem tervezett az 500-as hibakód.

> `{"type":"https://m2m.nav.gov.hu/technical-exception","title":"Technical exception occured in system: EBEV documentFileId: d4863763-a105-4595-8628-c5f9dba15b82","status":500,"detail":"feign.FeignException$InternalServerError: [500 Internal Server Error] during [POST] to [http://m2m-bizonylat-eloellenorzes/eloellenorzes] [BizonylatEloellenorzesServiceApi#bizonylatEloellenorzes(String,String,Mono)]: [{\"timestamp\":\"2024-10-22T12:16:22.872+00:00\",\"status\":500,\"error\":\"Internal Server Error\",\"trace\":\"jakarta.xml.ws.soap.SOAPFaultException: Connection prematurely closed BEFORE response\\n\\tat org.apach... (17375 bytes)]","instance":"/NavM2mDocument/documentService/Document","timestamp":"2024-10-22T12:16:22.878249243Z","application":"core-bizonylat","errorCategory":"Generic"}`

**Comment by Csonti-Piller** [comment]
Az itt leírt 500-as hibajelenséget kiváltó gyökér indokot nehezen találtuk, de sikerült felderíteni és megszüntetni, így ilyen hiba már nem jön létre a jövőben.
Ettől függetlenül továbbra is javasoljuk az 500-as hibakódra történő felkészülést a kliensprogramokban, amelyek kerüljenek bejelentésre, hogy kivizsgálhassuk őket.
